src/test/java/com/google/api/gax/grpc/CancellationTest.java

```diff
@@ -115,24 +120,36 @@ public void cancellationBeforeGetOnRetryingCallable() throws Exception {
     resultFuture.get();
   }
 
-  private static class LatchCountDownScheduler implements UnaryCallable.Scheduler {
-    private final ScheduledExecutorService executor;
-    private final CountDownLatch latch;
-
-    LatchCountDownScheduler(CountDownLatch latch) {
-      this.executor = new ScheduledThreadPoolExecutor(1);
-      this.latch = latch;
-    }
-
-    @Override
-    public ScheduledFuture<?> schedule(Runnable runnable, long delay, TimeUnit unit) {
-      latch.countDown();
-      return executor.schedule(runnable, delay, unit);
-    }
-
-    @Override
-    public List<Runnable> shutdownNow() {
-      return executor.shutdownNow();
+  private abstract static class LatchCountDownScheduler implements ScheduledExecutorService {
```

Comment:
This allows using SchedulerExecutorService directly in UnaryCallable and oteher callables, and avoids using the UnaryCallable.Scheduler. I believe that UnaryCallable.Scheduler was introduced just because of complications with mocking the full ScheduledExecutorService. Also, I had to avoid using UnaryCallable.Scheduler, because the retry call should be Grpc-independent.